### DISCUSSION TITLE
Beta geometric model fit taking forever!

### DISCUSSION THREAD

**Discussion by wfarrar-cil** [pymc-labs/pymc-marketing#369] [Q&A]
Hi guys,

I'm running a CLV analysis per instructions laid out in this article:

https://www.pymc-marketing.io/en/stable/notebooks/clv/clv_quickstart.html#predicting-spend-value-of-customers

I'm using my own dataset and it is in the exact required RFM format and has ~300,000 rows of data.

When I get to this step:

`bgm.fit()`
`bgm.fit_summary()`

The loading bar is saying it will take around 78 hours to run the code.

![image](https://github.com/pymc-labs/pymc-marketing/assets/113593843/d49efd6e-1562-4bb1-ac21-194864d847db)

Any ideas why it might be taking so long?

**Comment by ricardoV94** [comment]
You can try `fit(method="map")` instead of doing a full posterior fit. How fast does that run for you?
